gemcalibration/include/gem/calib/Calibration.h

```diff
@@ -0,0 +1,240 @@
+/** @file Calibration.h */
+
+#ifndef GEM_GEMCAL_CALIBRATION_H
+#define GEM_GEMCAL_CALIBRATION_H
+
+#include <string>
+#include <vector>
+
+#include "gem/base/GEMApplication.h"
+#include "gem/utils/Lock.h"
+#include "gem/utils/LockGuard.h"
+
+//#include "gem/daqmon/exception/Exception.h"
+#include "gem/utils/exception/Exception.h"
+//#include "gem/daqmon/DaqMonitor.h"
+
+#define NAMC 12
+#define NSHELF 2
+
+
+namespace gem {
+  namespace calib {
+
+    enum calType {NDEF, GBTPHASE, LATENCY, SCURVE, SBITARMDACSCAN, ARMDACSCAN, TRIMDAC, DACSCANV3, CALIBRATEARMDAC}; 
+    typedef enum calType calType_t;
+
+
+    enum dacScanType {CFG_CAL_DAC, CFG_BIAS_PRE_I_BIT, CFG_BIAS_PRE_I_BLCC, CFG_BIAS_PRE_I_BSF, CFG_BIAS_SH_I_BFCAS, CFG_BIAS_SH_I_BDIFF, CFG_BIAS_SD_I_BDIFF, CFG_BIAS_SD_I_BFCAS, CFG_BIAS_SD_I_BSF, CFG_BIAS_CFD_DAC_1, CFG_BIAS_CFD_DAC_2, CFG_HYST, CFG_THR_ARM_DAC, CFG_THR_ZCC_DAC, CFG_BIAS_PRE_VREF, CFG_VREF_ADC}; 
+    typedef enum dacScanType dacScanType_t;
+
+    
+    class Calibration : public gem::base::GEMApplication
+      {
+
+      public:
+        XDAQ_INSTANTIATOR();
+
+        Calibration(xdaq::ApplicationStub* s);
+
+        virtual ~Calibration();
+
+        virtual void init();
+
+        virtual void actionPerformed(xdata::Event& event);
+
+	// void startMonitoring();
+
+        //void stopMonitoring();
+
+        //std::string monitoringState(){return m_state;}
+
+        void stopAction(xgi::Input *in, xgi::Output *out)
+          throw (xgi::exception::Exception);
+
+        void resumeAction(xgi::Input *in, xgi::Output *out)
+          throw (xgi::exception::Exception);
+
+        void pauseAction(xgi::Input *in, xgi::Output *out)
+          throw (xgi::exception::Exception);
+	
+        void applyAction(xgi::Input *in, xgi::Output *out)
+          throw (xgi::exception::Exception);
+        
+        void setCalType(xgi::Input *in, xgi::Output *out)
+          throw (xgi::exception::Exception);
+
+        std::vector<Calibration*> v_gemcal;
```

Comment:
What is this? I.e., why does the `Calibration` class contain a vector of pointers to `Calibration` items?
Use smart pointers unless impossible due to compatibility with external libraries